File: 8KYU/Collatz Conjecture (3n+1)/solution.cs
public class Kata
{
  private static uint Next(uint n)
     => n % 2 == 0 ? n / 2 : 3 * n + 1;
  
  public static uint Hotpo(uint n)
  {
       uint c = 0;
       while(n != 1)
       {
          c++;
          n = Next(n);
       }
       return c;
  }
}

File: 8KYU/Collatz Conjecture (3n+1)/solution.cpp
unsigned int next(unsigned int n) {
  return n % 2 == 0 ? n / 2 : 3 * n + 1;
}
unsigned int hotpo(unsigned int n){
    unsigned int c = 0;
  
    while(n != 1) {
       c++;
       n = next(n);
    }
  
    return c;    
}